subscription_manager.go

```diff
@@ -2,24 +2,36 @@ package lnmux
 
 import (
 	"sync/atomic"
+	"time"
 
 	"github.com/lightningnetwork/lnd/lntypes"
 	"go.uber.org/zap"
 )
 
+// DefaultGracePeriodWithoutSubscribers is the default period during which the subscriberManager is waiting for a subscriber
+// to reconnect.
+const DefaultGracePeriodWithoutSubscribers = 30 * time.Second
+
 type subscriptionManager struct {
 	subscriptionHash    map[int]lntypes.Hash
 	acceptSubscriptions map[int]AcceptCallback
 	logger              *zap.SugaredLogger
 
-	nextSubscriberId uint64
+	nextSubscriberId              uint64
+	lastSubscriberRemovedAt       time.Time
+	gracePeriodWithoutSubscribers time.Duration
 }
 
-func newSubscriptionManager(logger *zap.SugaredLogger) *subscriptionManager {
+func newSubscriptionManager(logger *zap.SugaredLogger, gracePeriodWithoutSubscribers time.Duration) *subscriptionManager {
+	if gracePeriodWithoutSubscribers == 0 {
```

Comment:
I wouldn't do this. Defaults on various levels may get confusing. I'd just instantiate the invoice registry with the default value at a higher level.

Reply:
Done